delphi-frontend/src/main/java/org/sonar/plugins/communitydelphi/api/ast/TypeParameterNode.java

```diff
@@ -30,4 +31,12 @@ public interface TypeParameterNode extends DelphiNode {
   List<TypeReferenceNode> getTypeConstraintNodes();
 
   List<ConstraintNode> getConstraintNodes();
+
+  List<TypeParameter> getTypeParameters();
+
+  interface TypeParameter {
+    NameDeclarationNode getLocation();
+
+    TypeParameterType getType();
+  }
```

Comment:
This API already exists on `GenericDefinitionNode`.

Reply:
Yes - this addition is because `GenericDefinitionNode` combines all the type parameters for all `TypeParameterNode`s into a single list, which stops us from resolving both the type parameters and the name declarations in the correct order (left-to-right, not all name declarations followed by all type parameters). 

I even considered deprecating the version in `GenericDefinitionNode`.

Reply:
Alright, let's compromise for maximum misery - deprecated the old API altogether.

Reply:
As discussed, I've kept the old API and changed `TypeParameterNode` to use `GenericDefinitionNode.TypeParameter`.